src/Symfony/Component/HttpFoundation/JsonResponse.php

```diff
@@ -89,11 +89,16 @@ public static function fromJsonString($data = null, $status = 200, $headers = ar
     public function setCallback($callback = null)
     {
         if (null !== $callback) {
-            // taken from http://www.geekality.net/2011/08/03/valid-javascript-identifier/
-            $pattern = '/^[$_\p{L}][$_\p{L}\p{Mn}\p{Mc}\p{Nd}\p{Pc}\x{200C}\x{200D}]*+$/u';
```

Comment:
Removed the possessive `+` as this wont affect number of steps to match due `^$`.